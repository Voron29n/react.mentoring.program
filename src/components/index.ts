export * from './common/Button';
export * from './common/Logo';

export * from './containers/Header';
export * from './containers/Lightbox';
export * from './containers/MovieForm';
export * from './containers/MovieForm/MovieFormView';
export * from './containers/MovieList';
export * from './containers/MovieList/MovieListView';

export * from './service/DeleteMovie';
export * from './service/EditMovie';
export * from './service/AddMovie';
export * from './service/SearchMovie';

export * from './single/Confirm';
export * from './single/Congratulations';
export * from './single/DropDown';
export * from './single/DropDown/DropDownItem';
export * from './single/Footer';
export * from './single/FilterBar';
export * from './single/GenreBar';
export * from './single/GenreBar/GenreBarItem';
export * from './single/GenreCheckmarks';
export * from './single/MovieDetail';
export * from './single/MovieItem/MovieView';
export * from './single/MovieItem';
export * from './single/MovieInput';
export * from './single/MovieInput/MovieInputView/index';
export * from './single/SearchBar';
export * from './single/SortBar';
export * from './single/Spinner';
export * from './single/Redirect';

export * from './error/ErrorBoundary';
export * from './error/FetchErrorMessage';
export * from './error/UnknownPage';
