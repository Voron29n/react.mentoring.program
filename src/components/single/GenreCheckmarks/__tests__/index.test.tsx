import React from 'react';
import { render, RenderResult } from '@testing-library/react';
import { GenreCheckmarks } from 'components';
import { genreCheckmarksComponent } from './mocks';

const onChange = jest.fn();

describe('GenreCheckmarks component', () => {
  const { id, meta, placeholder, genreList, value } = genreCheckmarksComponent;
  let genreCheckmarksRender: RenderResult;

  beforeEach(() => {
    genreCheckmarksRender = render(
      <GenreCheckmarks
        onChange={onChange}
        id={id}
        meta={meta}
        placeholder={placeholder}
        genreList={genreList}
        value={value}
      />
    );
  });

  it('GenreCheckmarks snapshot', () => {
    expect(genreCheckmarksRender).toMatchSnapshot();
  });
});
