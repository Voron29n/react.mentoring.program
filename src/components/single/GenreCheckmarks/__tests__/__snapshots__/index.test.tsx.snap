// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreCheckmarks component GenreCheckmarks snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormControl-root css-19fbff2-MuiFormControl-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="test"
            class="MuiSelect-select MuiSelect-outlined MuiSelect-multiple MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="test"
            role="button"
            tabindex="0"
          >
            test1
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            placeholder="test placeholder"
            tabindex="-1"
            value="test1"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled css-1wc848c-MuiFormHelperText-root"
        >
          error
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormControl-root css-19fbff2-MuiFormControl-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="test"
          class="MuiSelect-select MuiSelect-outlined MuiSelect-multiple MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="test"
          role="button"
          tabindex="0"
        >
          test1
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          placeholder="test placeholder"
          tabindex="-1"
          value="test1"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled css-1wc848c-MuiFormHelperText-root"
      >
        error
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
